### PR TITLE
VM/VMSS: incorporate credentials validation logic used by portal

### LINKED ISSUE #1943: az vm create fails when user account name is too short
When running `az vm create` without specifying an admin-username and having a login name that is 'david' (I'm guessing anything 5 letters or less), your vm deployment fails.

`az vm create` should not use the current logged in username if it doesn't fit admin-username criteria.

### LINKED ISSUE #2010: VM Create prompting for admin username
My admin username was invalid and instead of throwing an error and exiting, it prompted me to enter it. It felt inconsistent with how usually, we simply throw an error.

```
$ az vm create -g debekoe-a4 -n mytestvm --image UbuntuLTS
Cannot use admin username: root. Admin username should be at least 6 characters and cannot be 'root'
Admin Username: ...
```

However, when I didn't have an SSH public key available, I received an error and there was no prompt:
```
An RSA key file or key value must be supplied to SSH Key Value. You can use --generate-ssh-keys to let CLI generate one for you.
```

If the admin username is invalid, should it not prompt and return an error such as below?
```
$ az vm create -g debekoe-a4 -n mytestvm --image UbuntuLTS
Cannot use admin username: root. Admin username should be at least 6 characters and cannot be 'root'. Specify admin username with --admin-username.
```

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #2010.  Fix #1943

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
...
+    if not is_linux and username.endswith('.'):
+        raise CLIError(win_err)
+    disallowed_user_names = [
+        "administrator", "admin", "user", "user1", "test", "user2",
+        "test1", "user3", "admin1", "1", "123", "a", "actuser", "adm",
+        "admin2", "aspnet", "backup", "console", "david", "guest", "john",
+        "owner", "root", "server", "sql", "support", "support_388945a0",
+        "sys", "test2", "test3", "user4", "user5"]
+    if username.lower() in disallowed_user_names:
+        raise CLIError('The specified admin user name is not allowed, as it uses reserved words. Try again with a different value')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Reserved words sounds like programming keywords like int, char, etc. Plus it's hard to say that "support_388945a0" is reserved. 😁  I would recommend something like "This user name "foo" meets the general requirements, but is specifically disallowed for this image. Please try a different value."

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
...
+                             contains_digit, contains_special_char] if x])
+    # pylint: disable=line-too-long
+    if count < 3:
+        raise CLIError('Password must have the 3 of the following: 1 lower case character, 1 upper case character, 1 number and 1 special character')
+    disallowed_passwords = [
+        "abc@123", "P@$$w0rd", "P@ssw0rd", "P@ssword123", "Pa$$word",
+        "pass@word1", "Password!", "Password1", "Password22", "iloveyou!"
+    ]
+    if password.lower() in disallowed_passwords:
+        raise CLIError('The specified password is not allowed')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: I'd recommend a similar error as above: that the password meets the general requirements, but it specifically disallowed.

File: src/command_modules/azure-cli-vm/tests/test_vm_actions.py [hunk 3]
```diff
@@ -65,3 +67,64 @@ def test_figure_out_storage_source(self):
         self.assertFalse(src_disk)
         self.assertFalse(src_snapshot)
         self.assertEqual(src_blob_uri, test_data)
+
+    def test_validate_admin_username_linux(self):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/tests/test_vm_actions.py]: 👍 for unit tests!

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 4]
```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
                 '/home/{}/.ssh/authorized_keys'.format(namespace.admin_username)
 
 
+def _validate_admin_username(username, is_linux):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Done

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 5]
```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
...
+    if not username:
+        return "admin user name can not be empty"
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Good catch. I missed this one when converted 'returning error strings' to throw

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: done

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: On a second thought, let me just remove this whole section. Those old banned passwords would not meet the length requirement anyway

**Pull request merged**